app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,59 @@
+class VendorsController < ApplicationController
+
+  def index
+    @vendors = Vendor.uniq.order(name: :asc)
+  end
+
+  def show
+    @vendors = Vendor.where(id: params[:id])
+  end
+
+  def new
+    @vendor = Vendor.new 
+    if @vendor.market_id
+      @current_market = @vendor.market.id
+    end
```

Comment:
I'm not sure if I'm missing something, but I wonder why do you need to check if the vendor has a market_id -  you just created a new one didn't assign it to it.  (and you don't use @current_market anywhere in the View of new vendor )